transmart-core-db/grails-app/services/org/transmartproject/db/clinical/MultidimensionalDataResourceService.groovy

```diff
@@ -509,6 +509,16 @@ class MultidimensionalDataResourceService extends AbstractDataResourceService im
         )
     }
 
+    @Override
+    @Transactional
+    void clearPatientSets() {
+        log.info 'Clearing patient sets ...'
+        QtPatientSetCollection.executeUpdate('delete QtPatientSetCollection')
```

Comment:
An alternative implementation:
```sql
update i2b2demodata.qt_query_result_instance set delete_flag = 'Y';
truncate i2b2demodata.qt_patient_set_collection;
```

Reply:
I tried truncating: 
```
def session = (StatelessSessionImpl) sessionFactory.openStatelessSession()
session.createSQLQuery('truncate table qt_patient_set_collection').executeUpdate()
```
but there were some issues with the table access.
regarding other tables than QtPatientSetCollection, is it better to set delete_flag than clear them?

Reply:
We could always change it to truncate if that is needed for performance.
Setting the delete_flag instead of actually deleting is a different way to achieve the same result, but it requires less changes.
The QtQueryMaster records do not have to be removed, they are reused in successive queries.

Reply:
So, doing the delete on QtQueryInstance, QtQueryResultInstance and QtPatientSetCollection should be sufficient.